tests/Oryx.Integration.Tests/LocalDockerTests/Fixtures/DbContainerFixtureBase.cs

```diff
@@ -30,12 +29,44 @@ public abstract class DbContainerFixtureBase : IDisposable
 
         public DbContainerFixtureBase()
         {
+            Console.WriteLine("Starting database container...");
             DbServerContainerName = RunDbServerContainer().ContainerName;
-            if (!WaitUntilDbServerIsUp())
+            Console.WriteLine("Done.");
+
+            bool insertedSampleData = false;
+            try
+            {
+                Console.WriteLine("Waiting for database server to be up...");
+                var isDbServerUp = WaitUntilDbServerIsUp();
+                if (isDbServerUp)
+                {
+                    Console.WriteLine("Done.");
+                    Console.WriteLine("Inserting sample data...");
+                    InsertSampleData();
+                    insertedSampleData = true;
+                    Console.WriteLine("Done.");
+                }
+                else
+                {
+                    Console.WriteLine("Database server was not up in time.");
+                }
+            }
+            catch (Exception ex)
             {
-                throw new Exception("Database not ready in time");
+                // Since Dispose method does not get called in case of exceptions, catch the exception here, log it
+                // and stop the container.
+                Console.WriteLine("Exception occurred while setting up the database container: " + ex.ToString());
+            }
+
+            if (!insertedSampleData)
+            {
+                Console.WriteLine("Stopping the container...");
```

Comment:
why don't you add this to the exception handler?

Reply:
Because a database might not be up in which case we would still want to stop the container just to be safe